### ISSUE TITLE
Scikit-fmm not building on mac os x with new numpy (1.18.0)

### ISSUE THREAD

**Issue by mrava87** [scikit-fmm/scikit-fmm#34]
Hi,
I have ``scikit-fmm`` as one of the dependencies of my project (http://github.com/equinor/pylops) and a couple of days ago my azure pipeline on mac os x stopped working with the following error when trying to build the wheels of scikit-fmm:

```
2019-12-29T22:22:39.9026630Z   Building wheel for scikit-fmm (setup.py): started
2019-12-29T22:22:44.5801050Z   Building wheel for scikit-fmm (setup.py): finished with status 'error'
2019-12-29T22:22:44.5805430Z   ERROR: Command errored out with exit status 1:
2019-12-29T22:22:44.5807000Z    command: /Users/runner/hostedtoolcache/Python/3.7.5/x64/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-8oodt2vp/scikit-fmm/setup.py'"'"'; __file__='"'"'/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-8oodt2vp/scikit-fmm/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-wheel-z495mm6n --python-tag cp37
2019-12-29T22:22:44.5808100Z        cwd: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-8oodt2vp/scikit-fmm/
2019-12-29T22:22:44.5808440Z   Complete output (43 lines):
2019-12-29T22:22:44.5808630Z   running bdist_wheel
2019-12-29T22:22:44.5809250Z   running build
2019-12-29T22:22:44.5809450Z   running config_cc
2019-12-29T22:22:44.5810170Z   unifing config_cc, config, build_clib, build_ext, build commands --compiler options
2019-12-29T22:22:44.5810470Z   running config_fc
2019-12-29T22:22:44.5811160Z   unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
2019-12-29T22:22:44.5811460Z   running build_src
2019-12-29T22:22:44.5811680Z   build_src
2019-12-29T22:22:44.5811900Z   building extension "skfmm.cfmm" sources
2019-12-29T22:22:44.5812140Z   building extension "skfmm.pheap" sources
2019-12-29T22:22:44.5812800Z   build_src: building npy-pkg config files
2019-12-29T22:22:44.5813090Z   running build_py
2019-12-29T22:22:44.5813320Z   creating build
2019-12-29T22:22:44.5813980Z   creating build/lib.macosx-10.13-x86_64-3.7
2019-12-29T22:22:44.5814700Z   creating build/lib.macosx-10.13-x86_64-3.7/skfmm
2019-12-29T22:22:44.5815420Z   copying skfmm/__init__.py -> build/lib.macosx-10.13-x86_64-3.7/skfmm
2019-12-29T22:22:44.5816240Z   copying skfmm/heap.py -> build/lib.macosx-10.13-x86_64-3.7/skfmm
2019-12-29T22:22:44.5817000Z   copying skfmm/pfmm.py -> build/lib.macosx-10.13-x86_64-3.7/skfmm
2019-12-29T22:22:44.5817780Z   copying skfmm/setup.py -> build/lib.macosx-10.13-x86_64-3.7/skfmm
2019-12-29T22:22:44.5818110Z   running build_ext
2019-12-29T22:22:44.5818360Z   customize UnixCCompiler
2019-12-29T22:22:44.5818630Z   customize UnixCCompiler using build_ext
2019-12-29T22:22:44.5818880Z   customize UnixCCompiler
2019-12-29T22:22:44.5819150Z   customize UnixCCompiler using build_ext
2019-12-29T22:22:44.5820080Z   building 'skfmm.cfmm' extension
2019-12-29T22:22:44.5820380Z   compiling C++ sources
2019-12-29T22:22:44.5821150Z   C compiler: g++ -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -std=c99
2019-12-29T22:22:44.5821500Z   
2019-12-29T22:22:44.5822200Z   creating build/temp.macosx-10.13-x86_64-3.7/skfmm
2019-12-29T22:22:44.5823120Z   compile options: '-Iskfmm -I/Users/runner/hostedtoolcache/Python/3.7.5/x64/lib/python3.7/site-packages/numpy/core/include -I/Users/runner/hostedtoolcache/Python/3.7.5/x64/include/python3.7m -c'
2019-12-29T22:22:44.5823500Z   g++: skfmm/fmm.cpp
2019-12-29T22:22:44.5823790Z   g++: skfmm/heap.cppg++: skfmm/base_marcher.cpp
2019-12-29T22:22:44.5824060Z   
2019-12-29T22:22:44.5824310Z   g++: skfmm/distance_marcher.cpp
2019-12-29T22:22:44.5825040Z   errorerror: : invalid argument '-std=c99' not allowed with 'C++'error:
2019-12-29T22:22:44.5825860Z   error: invalid argument '-std=c99' not allowed with 'C++'invalid argument '-std=c99' not allowed with 'C++'
2019-12-29T22:22:44.5826630Z   invalid argument '-std=c99' not allowed with 'C++'
2019-12-29T22:22:44.5826950Z   
2019-12-29T22:22:44.5827200Z   g++: skfmm/travel_time_marcher.cpp
2019-12-29T22:22:44.5827480Z   g++: skfmm/extension_velocity_marcher.cpp
2019-12-29T22:22:44.5828200Z   error: invalid argument '-std=c99' not allowed with 'C++'
2019-12-29T22:22:44.5828960Z   error: invalid argument '-std=c99' not allowed with 'C++'
2019-12-29T22:22:44.5830150Z   error: Command "g++ -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -std=c99 -Iskfmm -I/Users/runner/hostedtoolcache/Python/3.7.5/x64/lib/python3.7/site-packages/numpy/core/include -I/Users/runner/hostedtoolcache/Python/3.7.5/x64/include/python3.7m -c skfmm/base_marcher.cpp -o build/temp.macosx-10.13-x86_64-3.7/skfmm/base_marcher.o -MMD -MF build/temp.macosx-10.13-x86_64-3.7/skfmm/base_marcher.o.d" failed with exit status 1
2019-12-29T22:22:44.5831030Z   ----------------------------------------
2019-12-29T22:22:44.5831790Z   ERROR: Failed building wheel for scikit-fmm
2019-12-29T22:22:44.5832520Z   Running setup.py clean for scikit-fmm
```

I tried to reproduce this error locally on my mac and after some digging I found out that this one works fine

```
pip install numpy==1.17.3
pip install scikit-fmm --no-cache-dir
```

while this one leads to the above mentioned error:

```
pip install numpy==1.18.0
pip install scikit-fmm --no-cache-dir
```

Note that the latter is the latest `numpy` which you would also get if simply doing `pip install numpy` and so it will quickly be used by most people. As I see from your setup.py file that you use `numpy.distutils.misc_util` I wonder if some change in the new release of numpy could be responsible for this?

Thank you!

**Comment by jkfurtney** [comment]
Hi and thanks for getting in touch. This seems to be related to this issue:

https://github.com/scikit-image/scikit-image/issues/4369

and as far as I understand it has been fixed by the NumPy developers. Can you give it a try? I do not have and OS X system at the moment to confirm the fix.

**Comment by mrava87** [comment]
Hi,
Thanks a lot :) will give it a try today and let you know

**Comment by mrava87** [comment]
So far it doesn't work, but I think it makes sense as it will not be available until the release a patch (probably 1.18.1)... hopefully they will do it soon, i will keep an eye on the issue you mentioned and associated PR.

Along similar lines, I also noticed that you only release wheels for Windows on PyPy. When I build from linux or os x (either locally or on a CI machine) the following error arises - I tracked it back and it actually happened also with older versions of numpy. Basically as no wheels are present for those OS, the source is downloaded and wheels are built locally, but since you use ``numpy.distutils.misc_util`` in your build process an error is returned ``ModuleNotFoundError: No module named 'numpy'``.

```
2020-01-03T11:29:14.8508020Z Collecting scikit-fmm
2020-01-03T11:29:14.8611780Z   Downloading https://files.pythonhosted.org/packages/9a/a5/2bacbdbbde95598fdc36ded7968a2e590f51fcfcd81a68e5c783db131d80/scikit-fmm-2019.1.30.tar.gz (418kB) 
...
Building wheels for collected packages: scikit-fmm, sphinx-gallery, numpydoc, backcall, pandocfilters, pyrsistent
2020-01-03T11:29:26.3745000Z   Building wheel for scikit-fmm (setup.py): started
2020-01-03T11:29:26.5725920Z   Building wheel for scikit-fmm (setup.py): finished with status 'error'
2020-01-03T11:29:26.5728410Z   ERROR: Command errored out with exit status 1:
2020-01-03T11:29:26.5731210Z    command: /Users/runner/hostedtoolcache/Python/3.7.5/x64/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-dlkjnam8/scikit-fmm/setup.py'"'"'; __file__='"'"'/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-dlkjnam8/scikit-fmm/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-wheel-b2ksr9ft --python-tag cp37
2020-01-03T11:29:26.5733510Z        cwd: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-dlkjnam8/scikit-fmm/
2020-01-03T11:29:26.5734200Z   Complete output (7 lines):
2020-01-03T11:29:26.5734560Z   Traceback (most recent call last):
2020-01-03T11:29:26.5735210Z     File "<string>", line 1, in <module>
2020-01-03T11:29:26.5736500Z     File "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-dlkjnam8/scikit-fmm/setup.py", line 121, in <module>
2020-01-03T11:29:26.5737100Z       setup_package()
2020-01-03T11:29:26.5739220Z     File "/private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-install-dlkjnam8/scikit-fmm/setup.py", line 115, in setup_package
2020-01-03T11:29:26.5739790Z       from numpy.distutils.core import setup
2020-01-03T11:29:26.5740860Z   ModuleNotFoundError: No module named 'numpy'
2020-01-03T11:29:26.5742050Z   ----------------------------------------
2020-01-03T11:29:26.5743100Z   ERROR: Failed building wheel for scikit-fmm
2020-01-03T11:29:26.5744290Z   Running setup.py clean for scikit-fmm
```

After it automatically tries to rebuild from setup.py install and it succeeds in Unix (and OSX until the latest numpy):

```
2020-01-03T11:29:27.0236733Z     Running setup.py install for scikit-fmm: started
2020-01-03T11:29:36.0328614Z     Running setup.py install for scikit-fmm: finished with status 'done'
```

I tried to force numpy to be installed first, running the following sequence of commands:

```
python -m pip install --upgrade pip setuptools wheel django
pip install numpy
pip install -r requirements-dev.txt
pip install .
```

but I get the same error. You should be able to see the log here
https://dev.azure.com/MRAVA/PyLops/_build/results?buildId=299&view=results

I am not sure this is a big issue as things work fine anyways but I thought about mentioning it just in case you were not aware of it.

**Comment by jkfurtney** [comment]
Thanks for the note, I was unaware of this.

**Comment by mrava87** [comment]
I can confirm that the main problem is now solved, when using numpy 1.18.1 scikit-fmm builds successfully on OS X :)
